### DISCUSSION TITLE
✨ Do not auto-publish all modules to Daggerverse

### DISCUSSION THREAD

**Discussion by sagikazarmark** [dagger/dagger#12361] [Ideas]
### What are you trying to do?

Do not auto-publish the following modules to Daggerverse:

- Dev/test modules
- Untagged modules
- Project (application) modules
- Modules intended as a demo/bug reproduction
- Forks of modules

In my particular case:

- Test modules
  - https://daggerverse.dev/mod/github.com/sagikazarmark/daggerverse/archivist/tests@946ca10d1f567012e8c9d2f5b60a83e41c1c3bfa
  - https://daggerverse.dev/mod/github.com/sagikazarmark/daggerverse/arc/tests@946ca10d1f567012e8c9d2f5b60a83e41c1c3bfa
  - https://daggerverse.dev/mod/github.com/sagikazarmark/daggerverse/bats/tests@946ca10d1f567012e8c9d2f5b60a83e41c1c3bfa
  - https://daggerverse.dev/mod/github.com/sagikazarmark/daggerverse/checksum/tests@946ca10d1f567012e8c9d2f5b60a83e41c1c3bfa
  - https://daggerverse.dev/mod/github.com/sagikazarmark/daggerverse/gh/tests@946ca10d1f567012e8c9d2f5b60a83e41c1c3bfa
  - https://daggerverse.dev/mod/github.com/sagikazarmark/daggerverse/golangci-lint/tests@946ca10d1f567012e8c9d2f5b60a83e41c1c3bfa
  - https://daggerverse.dev/mod/github.com/sagikazarmark/daggerverse/go/tests@946ca10d1f567012e8c9d2f5b60a83e41c1c3bfa
  - https://daggerverse.dev/mod/github.com/sagikazarmark/daggerverse/helm-docs/tests@946ca10d1f567012e8c9d2f5b60a83e41c1c3bfa
  - https://daggerverse.dev/mod/github.com/sagikazarmark/daggerverse/helm/tests@946ca10d1f567012e8c9d2f5b60a83e41c1c3bfa
  - https://daggerverse.dev/mod/github.com/sagikazarmark/daggerverse/kafka/tests@946ca10d1f567012e8c9d2f5b60a83e41c1c3bfa
  - https://daggerverse.dev/mod/github.com/sagikazarmark/daggerverse/kustomize/tests@946ca10d1f567012e8c9d2f5b60a83e41c1c3bfa
  - https://daggerverse.dev/mod/github.com/sagikazarmark/daggerverse/python/tests@946ca10d1f567012e8c9d2f5b60a83e41c1c3bfa
  - https://daggerverse.dev/mod/github.com/sagikazarmark/daggerverse/quarto/tests@946ca10d1f567012e8c9d2f5b60a83e41c1c3bfa
  - https://daggerverse.dev/mod/github.com/sagikazarmark/daggerverse/registry-config/tests@946ca10d1f567012e8c9d2f5b60a83e41c1c3bfa
  - https://daggerverse.dev/mod/github.com/sagikazarmark/daggerverse/slsa-verifier/tests@946ca10d1f567012e8c9d2f5b60a83e41c1c3bfa
  - https://daggerverse.dev/mod/github.com/sagikazarmark/daggerverse/spectral/tests@946ca10d1f567012e8c9d2f5b60a83e41c1c3bfa
  - https://daggerverse.dev/mod/github.com/sagikazarmark/daggerverse/xk6/tests@946ca10d1f567012e8c9d2f5b60a83e41c1c3bfa
- Application modules
  - https://daggerverse.dev/mod/github.com/openmeterio/openmeter@209b194cf4caabc50a61e0891e486eb6453e1e25
- Demo modules
  - https://daggerverse.dev/mod/github.com/sagikazarmark/daggerverse/ssh-key-repro@34c3e978f32973d87c56c51442f90aa2ab193a12
- Forks
  - https://daggerverse.dev/mod/github.com/sagikazarmark/shykes-git@4f46bfde5735e17612316a30bc5a43c3eefd6da2
  - https://daggerverse.dev/mod/github.com/sagikazarmark/shykes-git/dev@4f46bfde5735e17612316a30bc5a43c3eefd6da2

### Why is this important to you?

Those modules create a lot of noise: https://daggerverse.dev/search?q=sagikazarmark

### How are you currently working around this?

I don't 🙈

**Comment by schmurfy** [comment]
I just started trying out the latest dagger to see what I could use it for and I agree that daggerverse is basically useless in that state, it looks more like a pile of trash than an actual module repository which is a shame...

I also have some issues with that auto publish feature as I work on private modules which should never get published, what was the idea behind it and why not rely on a manual publish system like any other platform ?
